src/core/control/tools/Selection.cpp

```diff
@@ -17,29 +17,51 @@ Selection::~Selection() = default;
 
 //////////////////////////////////////////////////////////
 
-RectSelection::RectSelection(double x, double y) {
+RectSelection::RectSelection(double x, double y, bool multiLayer) {
     bbox.addPoint(x, y);
     this->sx = x;
     this->sy = y;
     this->ex = x;
     this->ey = y;
+    this->multiLayer = multiLayer;
 }
 
 RectSelection::~RectSelection() = default;
 
-auto RectSelection::finalize(PageRef page) -> bool {
+auto RectSelection::finalize(PageRef page) -> size_t {
     this->page = page;
-
-    Layer* l = page->getSelectedLayer();
-    for (Element* e: l->getElements()) {
-        if (e->isInSelection(this)) {
-            this->selectedElements.push_back(e);
+    size_t layerId = 0;
+
+    if (multiLayer) {
+        for (int layerNo = page->getLayers()->size() - 1; layerNo >= 0; layerNo--) {
+            Layer* l = page->getLayers()->at(layerNo);
+            bool selectionOnLayer = false;
+            if (l->isVisible()) {
+                for (Element* e: l->getElements()) {
+                    if (e->isInSelection(this)) {
+                        this->selectedElements.push_back(e);
+                        selectionOnLayer = true;
+                    }
+                }
+            }
```

Comment:
For readability I would prefer less nesting.
```suggestion
            if (!l->isVisible()) {
                continue;
            }
            bool selectionOnLayer = false;
            for (Element* e: l->getElements()) {
                if (e->isInSelection(this)) {
                    this->selectedElements.push_back(e);
                    selectionOnLayer = true;
                }
            }
```